validation.go

```diff
@@ -209,8 +209,8 @@ func (v *subSchema) validateRecursive(currentSubSchema *subSchema, currentNode i
 
 			case bson.D:
 				// If this is a mixed type, it can be a DBRef { $ref, $db, $id }
-				isObjectType := v.types.Contains(TYPE_OBJECT) || v.bsonTypes.Contains(TYPE_OBJECT)
-				isMixedType := v.types.Contains(TYPE_MIXED) || v.bsonTypes.Contains(TYPE_MIXED)
+				isObjectType := currentSubSchema.types.Contains(TYPE_OBJECT) || currentSubSchema.bsonTypes.Contains(TYPE_OBJECT)
```

Comment:
😢  - I will add tests in the PR to make it use this

Reply:
I dont love that there arent tests in here. I suspect long term we should think about how we use / rely on this repo

Reply:
Yeah, do you think it's worth filing a ticket for this? Or perhaps something worth bringing up at a round table since it also concerns the core team as well